### ISSUE TITLE
EachValidator && ExistValidator

### ISSUE THREAD

**Issue by pkirill99** [yiisoft/yii2#10566]
Hi all

Yii 2.0.6

in model

```php
public function rules() {
        return [
            [['question', 'answers'], 'required'],
            [['question',], 'integer'],
            [['answers',], 'integer', 'min' => 1, 'on' => self::SCENARIO_SINGLE],
            [
                ['answers'],
                'exist',
                'targetClass'     => Answer::className(),
                'targetAttribute' => ['answers' => 'id', 'question' => 'question_id'],
                'on'              => self::SCENARIO_SINGLE,
            ],
            [['answers'], 'each', 'rule' => ['integer', 'min' => 1], 'on' => self::SCENARIO_MULTI],
            [
                ['answers'],
                'each',
                'rule' => [
                    'exist',
                    'targetClass'     => Answer::className(),
                    'targetAttribute' => ['answers' => 'id', 'question' => 'question_id'],
                ],
                'on'   => self::SCENARIO_MULTI,
            ],
        ];
    }
```

set scenario=SCENARIO_MULTI, run and see:

> The "targetAttribute" property must be configured as a string.

change rule for SCENARIO_MULTI, 
```php

            [
                ['answers'],
                'exist',
                'allowArray'      => true,
                'targetClass'     => Answer::className(),
                'targetAttribute' => ['answers' => 'id', 'question' => 'question_id'],
                'on'              => self::SCENARIO_MULTI,
            ],
```

run and see:
> The "targetAttribute" property must be configured as a string.

but in first time error in https://github.com/yiisoft/yii2/blob/master/framework/validators/ExistValidator.php#L131

and in second time error in https://github.com/yiisoft/yii2/blob/master/framework/validators/ExistValidator.php#L90

**Comment by Vovan-VE** [comment]
May be related to #9935.

**Comment by andrewnester** [comment]
I am not sure if it relates to #9935

Seems like behaviour of ExistValidator doesn't depend on context it's used (directly used or inside EachValidator).
For both cases we mention that string targetAttribute must be used

So if it will be confirmed as bug or improvement I suppose it will be connected only with ExistValidator

**Comment by pkirill99** [comment]
yes, ExistValidator has 'allowArray' as param, but it not work in 
https://github.com/yiisoft/yii2/blob/master/framework/validators/ExistValidator.php#L90

**Comment by andrewnester** [comment]
@pkirill99 you are absolutely right, I just fixed it and sent PR

**Comment by cebe** [comment]
As far as I see this is more a documentation issue. It should be made more clear how to use `allowArray`. Need to figure it out myself, so I can not give a solution right now...

**Comment by klimov-paul** [comment]
Should be resolved by commit 205f16811a6071ad5d1dbea4cba9fd8a3282f6c5

**Comment by edegaudenzi** [comment]
> As far as I see this is more a documentation issue. It should be made more clear how to use `allowArray`. Need to figure it out myself, so I can not give a solution right now...

Old one but stumbled today on this one. Wanted to give an example on how to use this validator with `'allowArray' => true` for my personal future reference but I hope to be useful to someone else, too. 

Say in your online store you want to create a Coupon that can be applied to one or more Products you are selling: assuming you don't want to have a middle table to hold the relations, the Coupon model will then need to have an attribute containing Product IDs as an array that can be used to validate if the Coupon can or cannot be applied to one or more Products.

Say your coupon attribute looks something like `->couponProductIDs = [456,332]`, then the validator would look something like:
`['couponProductIDs', 'exist', 'targetClass' => Product::class, 'targetAttribute' => 'id', 'allowArray' => true],`

and both `Product 456` and `Product 332` must exist to pass the validation.

Explained:
`couponProductIDs` - Is the Coupon array attributes containing an array of Product IDs
`exist` - name of the validator https://www.yiiframework.com/doc/api/2.0/yii-validators-existvalidator
`targetClass` - The pointed class to check the existence for, in this case is the Product class.
`targetAttribute` - The attribute of the pointed class to insist on for checking the existence for, in this case is Product->id
`allowArray` - Indicates that `couponProductIDs` can be an array of Product IDs and also, checks every ID be pointing to an existing Product.



\\\\//_
